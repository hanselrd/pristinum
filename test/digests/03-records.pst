struct Point:
    x @f64;
    y @f64;
end

func main() @void:
    p @Point := Point;
    p x .`` = 1.0;
    p y .`` = 2.0;
end