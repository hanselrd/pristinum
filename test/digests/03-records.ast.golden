PProgram
    [ PURecord
        ( RRecord RTStruct
            ( IDENT "Point" )
            [ BBind
                ( IDENT "x" ) TFloat64
            , BBind
                ( IDENT "y" ) TFloat64
            ]
        )
    , PUFunction
        ( FFunction
            ( IDENT "main" ) [] TVoid
            [ SBind
                ( BBind
                    ( IDENT "p" )
                    ( TIdent
                        ( IDENT "Point" )
                    )
                )
                ( EIdent
                    ( IDENT "Point" )
                )
            , SExpr
                ( EAssign
                    ( EAccess
                        ( EIdent
                            ( IDENT "p" )
                        )
                        ( EIdent
                            ( IDENT "x" )
                        )
                    )
                    ( EDouble 1.0 )
                )
            , SExpr
                ( EAssign
                    ( EAccess
                        ( EIdent
                            ( IDENT "p" )
                        )
                        ( EIdent
                            ( IDENT "y" )
                        )
                    )
                    ( EDouble 2.0 )
                )
            ]
        )
    ]